internal/utils/telegraf/emitter/baseemitter/baseemitter.go

```diff
@@ -0,0 +1,224 @@
+	// You can add metrics and events to exclude by name using
+	// ExcludeDatum(string) and ExcludeData(string).  You should look up
+	// excluded events and metrics using Excluded(string)bool
+	excluded map[string]bool
+}
+
+// AddTag adds a key/value pair to all measurement tags.  If a key conflicts
+// the key value pair in AddTag will override the original key on the
+// measurement
+func (B *BaseEmitter) AddTag(key string, val string) {
+	B.lock.Lock()
+	B.addTags[key] = val
+	B.lock.Unlock()
+}
+
+// AddTags adds a map of key value pairs to all measurement tags.  If a key
+// conflicts the key value pair in AddTags will override the original key on
+// the measurement.
+func (B *BaseEmitter) AddTags(tags map[string]string) {
+	for k, v := range tags {
+		B.AddTag(k, v)
+	}
+}
+
+// IncludeEvent a thread safe function for registering an event name to include
+// during emission. We disable all events by default because Telegraf has some
+// junk events.
+func (B *BaseEmitter) IncludeEvent(name string) {
+	B.lock.Lock()
+	B.included[name] = true
+	B.lock.Unlock()
+}
+
+// IncludeEvents a thread safe function for registering a list of event names to
+// include during emission. We disable all events by default because Telegraf
+// has some junk events.
+func (B *BaseEmitter) IncludeEvents(names []string) {
+	for _, name := range names {
+		B.IncludeEvent(name)
+	}
+}
+
+// Included - A thread safe function for checking if events should be included
+// during emission.  We disable all events by default because Telegraf has some
+// junk events.
+func (B *BaseEmitter) Included(name string) (included bool) {
+	B.lock.Lock()
+	included = B.included[name]
+	B.lock.Unlock()
+	return
+}
+
+// ExcludeDatum adds a name to the list of metrics and events to
+// exclude
+func (B *BaseEmitter) ExcludeDatum(name string) {
+	B.lock.Lock()
+	B.excluded[name] = true
+	B.lock.Unlock()
+}
+
+// ExcludeData adds a list of names the list of metrics and events
+// to exclude
+func (B *BaseEmitter) ExcludeData(names []string) {
+	for _, name := range names {
+		B.ExcludeDatum(name)
+	}
+}
+
+// Excluded - A thread safe function for checking if events or metrics should be
+// excluded from emission
+func (B *BaseEmitter) Excluded(name string) (excluded bool) {
+	B.lock.Lock()
+	excluded = B.excluded[name]
+	B.lock.Unlock()
+	return
+}
+
+// OmitTag adds a tag to the list of tags to remove from measurements
+func (B *BaseEmitter) OmitTag(tag string) {
+	B.lock.Lock()
+	B.omittedTags[tag] = true
+	B.lock.Unlock()
+}
+
+// OmitTags adds a list of tags the list of tags to remove from measurements
+func (B *BaseEmitter) OmitTags(tags []string) {
+	for _, tag := range tags {
+		B.OmitTag(tag)
+	}
+}
+
+// FilterTags - filter function for util.CloneAndFilterStringMapWithFunc()
+// that will exclude any tags that
+func (B *BaseEmitter) FilterTags(key string, value string) (include bool) {
+	B.lock.Lock()
+	include = B.omittedTags[key]
```

Comment:
This is a rather gratuitous use of named return values.  The normal idiom for this is simply:
```
B.lock.Lock()
defer B.lock.Unlock()
return B.omittedTags[key]
```

which is both shorter and clearer.

Reply:
Same thing above with the `Excluded` method.